src/GovUk.Education.ExploreEducationStatistics.Admin/Security/AuthorizationHandlers/PublishSpecificReleaseAuthorizationHandler.cs

```diff
@@ -37,10 +37,14 @@ public PublishSpecificReleaseAuthorizationHandler(IUserReleaseRoleRepository use
                 context.Succeed(requirement);
                 return;
             }
-
-            var releaseRoles = await _userReleaseRoleRepository.GetDistinctRolesByUserAndRelease(context.User.GetUserId(), release.Id);
-
-            if (ContainsApproverRole(releaseRoles))
+            
+            if (await _authorizationHandlerResourceRoleService
+                    .HasRolesOnPublicationOrRelease(
+                        context.User.GetUserId(),
+                        release.PublicationId,
+                        release.Id,
+                        ListOf(PublicationRole.Owner),
```

Comment:
Should PublicationRole.ReleaseApprover be included here, since ReleaseRole.Approver is included?

Reply:
yes it should!  Added, thanks!